http3/server.go

```diff
@@ -492,14 +527,25 @@ func (s *Server) handleConn(conn quic.Connection) error {
 	// Process all requests immediately.
 	// It's the client's responsibility to decide which requests are eligible for 0-RTT.
 	for {
-		str, datagrams, err := hconn.acceptStream(context.Background())
+		str, datagrams, err := hconn.acceptStream(s.closeCtx)
 		if err != nil {
+			// server is closed, send GOAWAY frame and return, since those requests will be rejected anyway
+			if errors.Is(err, context.Canceled) {
+				b = (&goawayFrame{
+					StreamID: lastID + 4,
+				}).Append(b[:0])
+				ctrlStr.Write(b)
```

Comment:
What happens if we're flow control blocked? Should we set a write deadline (5s maybe?), so we can guarantee that this function returns even if the peer is messing with our flow control window?